src/core/core_structs/simulations_stages.jl

```diff
@@ -1,24 +1,31 @@
 ######## Internal Simulation Object Structs ########
 abstract type AbstractStage end
 
-mutable struct _Stage <: AbstractStage
+mutable struct _Stage{M<:AbstractOperationModel} <: AbstractStage
     key::Int64
-    model::OperationModel
+    reference::ModelReference
+    op_model::Type{M}
+    sys::PSY.System
+    canonical::CanonicalModel
+    optimizer::JuMP.OptimizerFactory
     executions::Int64
     execution_count::Int64
-    optimizer::String
     chronology_ref::Dict{Int64, Type{<:Chronology}}
     ini_cond_chron::Union{Type{<:Chronology}, Nothing}
     cache::Dict{Type{<:AbstractCache}, AbstractCache}
 
     function _Stage(key::Int64,
-                   model::OperationModel,
-                   executions::Int64,
-                   chronology_ref::Dict{Int64, Type{<:Chronology}},
-                   cache::Vector{<:AbstractCache})
+                    reference::ModelReference,
+                    op_model::Type{M},
+                    sys::PSY.System,
+                    canonical::CanonicalModel,
```

Comment:
An op_model has a canonical_model, right? If so, would it be better to only pass op_model here?

Reply:
This has changes since the review